emsdk.py

```diff
@@ -1450,6 +1450,78 @@ def find_emscripten_root(active_tools):
   return root
 
 
+def fetch_nightly_node_versions():
+  url = "https://nodejs.org/download/nightly/"
+  with urlopen(url) as response:
+    html = response.read().decode("utf-8")
+
+  # Regex to capture href values like v7.0.0-nightly2016080175c6d9dd95/
+  pattern = re.compile(r'<a href="(v[0-9]+\.[0-9]+\.[0-9]+-nightly[0-9a-f]+)/">')
+  matches = pattern.findall(html)
+  return matches
+
+
+def dir_installed_nightly_node_versions():
+  path = os.path.abspath('node')
+  return [name for name in os.listdir(path) if os.path.isdir(os.path.join(path, name)) and name.startswith("nightly-")]
+
+
+def extract_newest_node_nightly_version(versions):
+  def parse(v):
+    # example: v7.0.0-nightly2016080175c6d9dd95
+    m = re.match(r'v(\d+)\.(\d+)\.(\d+)-nightly(\d+)', v)
+    if m:
+      major, minor, patch, nightly = m.groups()
+      return [int(major), int(minor), int(patch), int(nightly)]
+    else:
+      return []
+
+  try:
+    return max(versions, key=lambda v: parse(v))
+  except Exception:
+    return None
+
+
+def build_node_nightly(tool):
```

Comment:
Is this actually building node from source?  It looks like its just download it, so maybe this should be called "download_...` to `fetch_...`?